### PR TITLE
Implemented BadRequest response and profileShare

### PR DESCRIPTION
**Pull request by VenuAjitesh**
- Added WebClientResponseException.BadRequest for the response of POST request
- Implemented Profile share functionality for HIP

### CODE REVIEW

**Review by NHA-ABDM — APPROVED** [review]

File: src/main/java/com/nha/abdm/wrapper/common/responses/NestedErrorResponse.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class NestedErrorResponse {
```
Comment (atulai-sg) [inline src/main/java/com/nha/abdm/wrapper/common/responses/NestedErrorResponse.java:13]: please rename this to ErrorWrapper

File: src/main/java/com/nha/abdm/wrapper/hip/hrp/share/ProfileShareService.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            profileShare.getProfile().getPatient().getYearOfBirth()
+                + "-"
+                + profileShare.getProfile().getPatient().getMonthOfBirth()
+                + "-"
+                + profileShare.getProfile().getPatient().getDayOfBirth());
+        patient.setPatientDisplay(profileShare.getProfile().getPatient().getName());
+        patient.setPatientMobile(
+            profileShare.getProfile().getPatient().getIdentifiers().get(0).getValue());
+        patient.setEntity(profileShare.getProfile().getHipCode());
+        patientRepo.save(patient);
```
Comment (atulai-sg) [inline src/main/java/com/nha/abdm/wrapper/hip/hrp/share/ProfileShareService.java:89]: repo save is not recommended as it inserts duplicates. please use PatientService's upsertPatients method

File: src/main/java/com/nha/abdm/wrapper/hip/hrp/share/TokenNumberGenerator.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.stereotype.Component;
+
+@Component
+public class TokenNumberGenerator {
+
+  private AtomicInteger counter = new AtomicInteger(0);
+  private LocalDate currentDate = LocalDate.now();
+
+  @Scheduled(cron = "0 0 0 * * *")
+  public void resetTokenCount() {
```
Comment (atulai-sg) [inline src/main/java/com/nha/abdm/wrapper/hip/hrp/share/TokenNumberGenerator.java:16]: can you explain the logic behind generation of token? also, what does that cron expression evaluate to?

File: src/main/java/com/nha/abdm/wrapper/hip/hrp/share/ProfileShareService.java (hunk 2 above)
Comment (VenuAjitesh) [reply src/main/java/com/nha/abdm/wrapper/hip/hrp/share/ProfileShareService.java:89]: After creating the object, adding the patient into db via patientService.upsertPatients(Collections.singletonList(patient));

File: src/main/java/com/nha/abdm/wrapper/hip/hrp/share/TokenNumberGenerator.java (hunk 3 above)
Comment (VenuAjitesh) [reply src/main/java/com/nha/abdm/wrapper/hip/hrp/share/TokenNumberGenerator.java:16]: the 0 0 0 * * * indicates.. seconds minutes hours days months years, so according to it the first 3 zeros indicate 12 am and the * indicate every day of month.

**Pull request merged**